### DISCUSSION TITLE
希望开发dn42全自动版本

### DISCUSSION THREAD

**Discussion by famuserpro** [nxtrace/NTrace-core#76] [Ideas]
现在dn42需要手动添加ip段和rdns记录，有点麻烦，希望能够内置在里面，完全自动化

**Comment by sjlleo** [comment]
你好，

有点我不是很理解，DN42都是内网，除非穿透，公网是无法访问的，我们可并没有能力仅根据一个延迟来判断地理位置哦。

**Comment by Hawkins-Sherpherd** [comment]
你好，我写了一个用 dn42 registry 数据生成 geoip 数据的脚本，但是暂时没有rdns的：
https://github.com/Hawkins-Sherpherd/geoip_primary_gen_dn42

**Comment by tsosunchia** [comment]
你好，我并不了解DN42，如有需要欢迎提交PR

Hawkins Sherpherd ***@***.***>于2023年12月16日 周六下午2:36写道：

**Comment by Cryolitia** [comment]
我的想法是，在没有外部数据库的情况下，不一定要求nexttrace能够显示DN42节点的真实物理位置。DN42有一个注册表位于 https://git.dn42.dev/dn42/registry ，里面有诸如ASN和名字的信息，很多部署了DNS的自治域也可以通过IP反查域名。可以让DN42网络中trace的时候展示这些信息，至少比光秃秃的RFC4193好太多了

**Reply by tsosunchia** [reply]
这是个很好的想法 可以pr一下 我不懂dn42

**Reply by Cryolitia** [reply]
https://github.com/Cryolitia/Next-Trace-DN42-Feeder

<img width="1734" height="641" alt="image" src="https://github.com/user-attachments/assets/d9960750-62ac-42d0-94f1-064e8375d66f" />

我这边写了一个简单的脚本来生成信息，需要 PR https://github.com/nxtrace/NTrace-V1/pull/119 的修复。

可以考虑一下用action定期自动生成，同时在不指定参数的时候自动拉取，在指定参数的时候调用本地文件？

另外对齐可能还需要修一下（小声

**Reply by Cryolitia** [reply]
更进一步的，我们应该可以开发一个每日自动构建更新的类PTR格式的静态API出来，扔到cloudflare或者github pages上。

例如，对于 `172.123.234.1` ，从 `https://foo.org/172/123/234/1` 一路递归查找到根页面或者找到结果。

**Reply by tsosunchia** [reply]
> 更进一步的，我们应该可以开发一个每日自动构建更新的类PTR格式的静态API出来，扔到cloudflare或者github pages上。
> 
> 例如，对于 `172.123.234.1` ，从 `https://foo.org/172/123/234/1` 一路递归查找到根页面或者找到结果。

您的idea确实很好，但是我不懂DN42，做该功能的author目前AFK了，所以该DN42功能处于无人维护的状态，您需要什么功能可能需要DIY（自行PR）了。
